### PR TITLE
add M_PI macro for windows in test_component_parser.cpp

### PR DESCRIPTION
**Pull request by dzywater**

### CODE REVIEW

File: hardware_interface/test/test_component_parser.cpp [hunk 1]
```diff
@@ -20,6 +20,10 @@
 #include "ros2_control_test_assets/components_urdfs.hpp"
 #include "ros2_control_test_assets/descriptions.hpp"
 
+#ifndef M_PI
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply hardware_interface/test/test_component_parser.cpp:23]: @dzywater you forgot to reply to this comment before resolving it

File: hardware_interface/test/test_component_parser.cpp [hunk 2]
```diff
@@ -20,6 +20,10 @@
...
+#define M_PI 3.1415926535897932384626433832795
+#endif
```
Comment (destogl) [inline hardware_interface/test/test_component_parser.cpp:25]: And probably here something line `#ifdef WIN_64`
```suggestion
#define _USE_MATH_DEFINES
#include <cmath>
```

Comment (ken551) [reply hardware_interface/test/test_component_parser.cpp:25]: hello.
It seems that `<cmath>` should be included before `gmock/gmock.h`.
In my environment, including `cmath` after `gmock/gmock.h` didn't work but before it worked.

File: hardware_interface/test/test_component_parser.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bmagyar) [reply hardware_interface/test/test_component_parser.cpp:23]: how about we make this hack explicit 

```suggestion
#ifdef _WIN32
```

**Review by destogl — APPROVED** [review]
We can go on with this explicit windows flag.

**Pull request merged**